internal/command/iam_idp_auth_connector_config_model.go

```diff
@@ -0,0 +1,112 @@
+package command
+
+import (
+	"context"
+
+	"github.com/caos/zitadel/internal/domain"
+	"github.com/caos/zitadel/internal/eventstore"
+	"github.com/caos/zitadel/internal/repository/iam"
+	"github.com/caos/zitadel/internal/repository/idpconfig"
+)
+
+type IAMIDPAuthConnectorConfigWriteModel struct {
+	AuthConnectorConfigWriteModel
+}
+
+func NewIAMIDPAuthConnectorConfigWriteModel(idpConfigID string) *IAMIDPAuthConnectorConfigWriteModel {
+	return &IAMIDPAuthConnectorConfigWriteModel{
+		AuthConnectorConfigWriteModel{
+			WriteModel: eventstore.WriteModel{
+				AggregateID:   domain.IAMID,
+				ResourceOwner: domain.IAMID,
+			},
+			IDPConfigID: idpConfigID,
+		},
+	}
+}
+
+func (wm *IAMIDPAuthConnectorConfigWriteModel) AppendEvents(events ...eventstore.EventReader) {
+	for _, event := range events {
+		switch e := event.(type) {
+		case *iam.IDPAuthConnectorConfigAddedEvent:
+			if wm.IDPConfigID != e.IDPConfigID {
+				continue
+			}
+			wm.AuthConnectorConfigWriteModel.AppendEvents(&e.AuthConnectorConfigAddedEvent)
+		case *iam.IDPAuthConnectorConfigChangedEvent:
+			if wm.IDPConfigID != e.IDPConfigID {
+				continue
+			}
+			wm.AuthConnectorConfigWriteModel.AppendEvents(&e.AuthConnectorConfigChangedEvent)
+		case *iam.IDPConfigReactivatedEvent:
+			if wm.IDPConfigID != e.ConfigID {
+				continue
+			}
+			wm.AuthConnectorConfigWriteModel.AppendEvents(&e.IDPConfigReactivatedEvent)
+		case *iam.IDPConfigDeactivatedEvent:
+			if wm.IDPConfigID != e.ConfigID {
+				continue
+			}
+			wm.AuthConnectorConfigWriteModel.AppendEvents(&e.IDPConfigDeactivatedEvent)
+		case *iam.IDPConfigRemovedEvent:
+			if wm.IDPConfigID != e.ConfigID {
+				continue
+			}
+			wm.AuthConnectorConfigWriteModel.AppendEvents(&e.IDPConfigRemovedEvent)
+		default:
+			wm.AuthConnectorConfigWriteModel.AppendEvents(e)
```

Comment:
you're right... copy/paste from oidc model :)
will remove it there as well and simplify reduce:
```
func (wm *IAMIDPAuthConnectorConfigWriteModel) Reduce() error {
	return wm.AuthConnectorConfigWriteModel.Reduce()
}
```
instead of:
```
func (wm *IAMIDPAuthConnectorConfigWriteModel) Reduce() error {
	if err := wm.AuthConnectorConfigWriteModel.Reduce(); err != nil {
		return err
	}
	return wm.WriteModel.Reduce()
}
```